src/overlays/actors/ovl_En_Dog/z_en_dog.c

```diff
@@ -22,41 +28,579 @@ const ActorInit En_Dog_InitVars =
     (ActorFunc)EnDog_Update,
     (ActorFunc)EnDog_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dog/func_809FAFD0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dog/func_809FB038.s")
+static ColliderCylinderInit cylinderInit =
+{
+    0x06, 0x00, 0x09, 0x39,
+    0x10, 0x01, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00,
+    0x00000000,
+    0x00, 0x00, 0x00, 0x00,
+    0xFFCFFFFF,
+    0x00, 0x00, 0x00, 0x00,
+    0x00, 0x01, 0x01, 0x00,
+
+    0x0010,
+    0x0014,
+    0x0000,
+    0x0000, 0x0000, 0x0000,
+};
+
+// There are multiple versions of sub98 init structs
+// Keeping these local until z_collision_check is decompiled
+typedef struct
+{
+    /* 0x00 */ u8 health;
+    /* 0x02 */ s16 unk_10;
+    /* 0x04 */ s16 unk_12;
+    /* 0x06 */ u16 unk_14;
+    /* 0x08 */ u8 mass;
+}sub98Init;
+
+static sub98Init sub98Data = 
+{
+    0x00,   //health
+    0x0000, //unk_10
+    0x0000, //unk_12
+    0x0000, //unk_14
+    0x32,   //mass
+};
+
+static struct_80034EC0_Entry animations[]=
+{
+    {0x06001368, 1.0f, 0.0f, -1.0f, 0x00,  0.0f},
+    {0x06001368, 1.0f, 0.0f, -1.0f, 0x00, -6.0f}, 
+    {0x06000D78, 1.0f, 0.0f, -1.0f, 0x00, -6.0f}, 
+    {0x06000278, 1.0f, 0.0f, -1.0f, 0x00, -6.0f}, 
+    {0x06001150, 1.0f, 0.0f,  4.0f, 0x02, -6.0f}, 
+    {0x06001150, 1.0f, 5.0f,  25.0f,0x04, -6.0f}, 
+    {0x06000928, 1.0f, 0.0f,  6.0f, 0x02, -6.0f}, 
+    {0x06000C28, 1.0f, 0.0f, -1.0f, 0x00, -6.0f},
+};
+
+// Bandaid fix for a lw vs lh issue in EnDog_FollowPath. Roman will look at it later.
+typedef union
+{
+    /* 0x00 */ s32 entry;
+    struct 
+    {
+        s16 unk_0;
+        s16 unk_2;
+    };
+}s16ArrEntry;
+
+typedef enum
+{
+    /* 0x00 */ DOG_WALK,
+    /* 0x01 */ DOG_RUN,
+    /* 0x02 */ DOG_BARK,
+    /* 0x03 */ DOG_SIT,
+    /* 0x04 */ DOG_SIT_2,
+    /* 0x05 */ DOG_BOW,
+    /* 0x06 */ DOG_BOW_2,
+} DogBehavior;
+
+extern UNK_PTR D_06007290;
+extern UNK_PTR D_06001368;
+extern UNK_PTR D_06000D78;
+extern UNK_PTR D_06000278;
+
+static void EnDog_WalkSFX(EnDog* this)
```

Comment:
I suggest naming these with "Play" to avoid any confusion with action functions.
```suggestion
static void EnDog_PlayWalkSFX(EnDog* this)
```